### PR TITLE
Changes to the misc section of the DG + updating Ui

### PR DESCRIPTION
**Pull request by woojiahao**
- Homepage preview: https://woojiahao.com/tp/
- DG preview: https://woojiahao.com/tp/DeveloperGuide.html

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -3,39 +3,97 @@ layout: page
...
+- View their total expenditure by the month/year
+
+### Good-to-haves
+
+Every user can...
+
+- Visualise their spending habits
+- Divide a group spending and tag friends involved in the spending
+- View a price list comparison of all available shops near NUS
+- Set recurring transactions
```
Comment (elhy1999) [inline docs/DeveloperGuide.md:41]: Just a suggestion to improve this segment. Do you think it'll be better if we add tick marks next to these "Must-haves" and "Good-to-haves" to indicate which of these have been implemented? Nothing wrong with the current bullet-points though! 😄

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -3,39 +3,97 @@ layout: page
...
-- Tabulate total expenditure  - Sum of all expenses & remaining balance
+
+[//]: # (TODO: Link to each DG/UG entry)
+
+- Add transactions
+- Delete transactions
+- Clear transactions
+- Edit transactions
+- Find transaction
+- Get total expenditure
```
Comment (elhy1999) [inline docs/DeveloperGuide.md:58]: Minor grammatical fix

```suggestion
- Add transaction
- Delete transaction
- Clear transactions
- Edit transaction
- Find transactions
- Get total expenditure
```

**Review by elhy1999 — CHANGES_REQUESTED** [review]
Thanks for updating the DG @woojiahao! Just have some very minor suggestions to make things a little clearer which you can consider

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (jamz903) [reply docs/DeveloperGuide.md:58]: Would be better if _Clear transactions_ was changed to _Clear all transactions_, and _Find transaction_ to _Find transaction(s)_.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -3,39 +3,97 @@ layout: page
...
+
+[//]: # (TODO: Maybe don't need to talk about this since it's part of "add transaction" now and Lip Wei can elaborate on this separately)
 
 #### Categorize Finances
+
+This feature serves as an extension to adding a transaction as categories provide finer details about a transaction.
+
 - Add tags – provide category of transaction
 - Edit tags - change category of transaction
 - Remove tags - delete category of transaction
```
[the remark this answers is not in the document]
Comment (jamz903) [reply docs/DeveloperGuide.md:74]: I agree with this! It should be edited for more clarity.

```suggestion
This feature serves as an extension to adding a transaction as categories provide finer details about a transaction. Using the `edit transaction` command, users can now:
- Add categories to a transaction
- Edit existing categories of a transaction
- Delete existing categories of a transaction
```

**Review by jamz903 — COMMENTED** [review]
looks good overall! suggested some changes for you to consider 👍

File: docs/DeveloperGuide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (woojiahao) [reply docs/DeveloperGuide.md:74]: But we don't directly edit/remove them right? Like everything is done through `add_transaction` or `edit_transaction`

That's why I'm unsure if we should categorise them

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (woojiahao) [reply docs/DeveloperGuide.md:41]: I believe we have implemented all of the must-haves but only one of the good-to-haves. I'm not sure if having ticks will make them think we're missing features...

File: docs/DeveloperGuide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (woojiahao) [reply docs/DeveloperGuide.md:74]: Ah I just saw @jamz903 's suggestion, I'll incorporate that

**Review by jamz903 — DISMISSED** [review]
lgtm!

**Review by elhy1999 — APPROVED** [review]
LGTM!

**Pull request merged**